sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -85,6 +90,43 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
     new File(tmpDataDir, name).getCanonicalPath
   }
 
+  private def getFileFromUrl(urlString: String, targetDir: String, filename: String): Boolean = {
+    val conf = new SparkConf
+    val securityManager = new SecurityManager(conf)
+    val hadoopConf = new Configuration
+
+    val outDir = new File(targetDir)
+    if (!outDir.exists()) {
+      outDir.mkdirs()
+    }
+
+    try {
+      val result = Utils.doFetchFile(urlString, outDir, filename, conf, securityManager, hadoopConf)
+      result.exists()
+    } catch {
+      case ex: Exception => logWarning("Could not get file from url " + urlString + ": "
+        + ex.getMessage)
+        false
+    }
+  }
+
+  private def getStringFromUrl(urlString: String, encoding: String = "UTF-8"): String = {
+    val outDir = Files.createTempDirectory("string-")
+    val filename = "string-out.txt"
+
+    if (!getFileFromUrl(urlString, outDir.toString, filename)) {
+      throw new IOException("Could not get string from url " + urlString)
+    }
+
+    val contentFile = new File(outDir.toFile, filename)
+    try {
+      Source.fromFile(contentFile)(Codec(encoding)).mkString
```

Comment:
This is ok, but there's an easier / cleaner API for this. Look around for calls like this:

    Files.toString(file, StandardCharsets.UTF_8);